extensions/Lily/MoreEvents.js

```diff
@@ -546,6 +602,16 @@
       return util.ioQuery("keyboard", "getKeyIsDown", [key]);
     }
 
+    whenbroadcastreceived(args, util) {
+      const broadcastOption = Scratch.Cast.toString(args.BROADCAST_OPTION);
+      return broadcastOption === runtime.lmsLastBroadcastSent;

```

Comment:
Here we can't use `util.thread.broadcastName`, as`startHats` runs the hat code straight away, meaning it runs before broadcastName can even be assigned, which is pretty annoying